torchdynamo/utils.py

```diff
@@ -37,6 +37,44 @@
 
 log = logging.getLogger(__name__)
 
+# profiling compilation time
+compilation_metrics = collections.OrderedDict()
+@contextlib.contextmanager
+def timed_compile(key):
```

Comment:
1) Do we always want this enabled? This feels a bit like a debug /profiling thing.

2) The usage feels a little off:

```
with timed_compile("compile_fx_aot"):
    with timed_compile("normalize_ir"):
        with overrides.patch_functions():
            model_ = normalize_ir(model_, example_inputs_)
            model_ = overrides.replace_fx(model_)
        num_example_inputs = len(example_inputs_)
        cudagraphs = BoxedBool(config.triton.cudagraphs)
```

Why are we nesting two? one inside `compile_fx_aot` and one outside `normalize_ir`? This feels like it should be a decorator `@timed_compile` that just takes the fn name and uses that as the key.

3) The notion of `compile` here is lost on me - if we follow (1) and (2) and make this a decorator that no-ops when the appropriate debug flag is not set, can we just call it @time

Maybe something like:
```
def time(func):
    def wrap(*args, **kwargs):
        t0 = time.time()
        r = func(*args, **kwargs)
        compilation_metrics[func.__name__] = t0 - time.time()
        return r
```

Reply:
Further thought - if we do an @time, we should consider @profile also that runs cprofile, perhaps?